src/patch/dll_collision_viewer.zig

```diff
@@ -0,0 +1,351 @@
+        mi.MenuItemToggle(&QuickRaceMenu.item_cull_backfaces.input_converted, "Cull backfaces"),
+        mi.MenuItemToggle(&QuickRaceMenu.item_cull_backfaces.input_converted, "Cull backfaces"),
+        mi.MenuItemRange(&QuickRaceMenu.item_depth_bias.input_converted, "Depth bias", 0, 100, false),
+    };
+
+    var item_enabled = ConvertedMenuItem{ .input_bool = &state.enabled };
+    var item_show_visual_mesh = ConvertedMenuItem{ .input_bool = &state.settings.show_visual_mesh };
+    var item_mesh_opacity = ConvertedMenuItem{ .input_float = &state.settings.collision_mesh_opacity };
+    var item_mesh_brightness = ConvertedMenuItem{ .input_float = &state.settings.collision_mesh_brightness };
+    var item_line_opacity = ConvertedMenuItem{ .input_float = &state.settings.collision_line_opacity };
+    var item_line_brightness = ConvertedMenuItem{ .input_float = &state.settings.collision_line_brightness };
+    var item_depth_test = ConvertedMenuItem{ .input_bool = &state.settings.depth_test };
+    var item_cull_backfaces = ConvertedMenuItem{ .input_bool = &state.settings.cull_backfaces };
+    var item_depth_bias = ConvertedMenuItem{ .input_float = &state.depth_bias };
+
+    var all_items = [_]*ConvertedMenuItem{
+        &item_enabled,
+        &item_show_visual_mesh,
+        &item_mesh_opacity,
+        &item_mesh_brightness,
+        &item_line_opacity,
+        &item_line_brightness,
+        &item_depth_test,
+        &item_cull_backfaces,
+        &item_depth_bias,
+    };
+
+    fn init() void {
+        initialized = true;
+    }
+
+    fn open() void {
+        if (!gf.GameFreezeEnable(menu_key)) return;
+        //rf.swrSound_PlaySound(78, 6, 0.25, 1.0, 0);
+        data.idx = 0;
+        menu_active = true;
+    }
+
+    fn close() void {
+        if (!gf.GameFreezeDisable(menu_key)) return;
+        rf.swrSound_PlaySound(77, 6, 0.25, 1.0, 0);
+        _ = mem.write(rc.ADDR_PAUSE_STATE, u8, 3);
+        menu_active = false;
+    }
+
+    fn update() void {
+        if (gs.in_race == .JustOn)
+            init();
+
+        if (!initialized or !gs.practice_mode or !gs.in_race.on()) {
+            state.enabled = false;
+            return;
+        }
+
+        if (input_enable.gets() == .JustOn)
+            state.enabled = !state.enabled;
+
+        if (input_show_visual_mesh.gets() == .JustOn)
+            state.settings.show_visual_mesh = !state.settings.show_visual_mesh;
+
+        if (input_pause.gets() == .JustOn) {
+            if (menu_active) close() else open();
+        }
+
+        if (menu_active) {
+            var current_preset : i32 = @intCast(presets.len);
+            for (presets, 0..) |preset, index| {
+                if (std.meta.eql(preset,state.settings))
+                    current_preset = @intCast(index);
+            }
+            // QuickRaceMenuItems[2].max = 1; // if (current_preset == presets.len) presets.len+1 else presets.len;
```

Comment:
i have a zig question here... i want to change the length of the "preset_names", its either 4 or 5. if i add a line like that here i get a "dependency loop detected" error 😅

Reply:
i have a slightly different solution: i renamed "[modified]" to "[custom]", it stores your settings if you modify a preset. this makes it possible to switch between presets and your modified version without losing the settings. i think this is an acceptable solution.